x-pack/plugins/global_search_providers/server/providers/saved_objects/provider.ts

```diff
@@ -51,6 +51,15 @@ export const createSavedObjectsResultProvider = (): GlobalSearchResultProvider =
         map(([res, cap]) => mapToResults(res.saved_objects, typeRegistry, cap))
       );
     },
+    getSearchableTypes: ({ core }) => {
+      const {
+        savedObjects: { typeRegistry },
+      } = core;
+      return typeRegistry
+        .getVisibleTypes()
+        .filter((type) => type.management?.defaultSearchField && type.management?.getInAppUrl)
+        .map((type) => type.name);
```

Comment:
Should we DRY this up with lines 26-29 to ensure the implementations of these two functions are consistent?